### PR TITLE
PEER-231 Form Feedback and Validation

### PR DESCRIPTION
**Pull request by SeeuSim**

### CODE REVIEW

**Comment by SeeuSim** [comment]
Validation messages:

image

File: frontend/src/routes/login/logic.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { useForm } from 'react-hook-form';
+
+export const loginFormSchema = z.object({
+  username: z.string().min(1),
+  password: z.string().min(1),
+});
+
+type ILoginFormSchema = z.infer<typeof loginFormSchema>;
+
+const onSubmit = (_formData: ILoginFormSchema) => {
```
Comment (TabrizPlv) [inline frontend/src/routes/login/logic.ts:14]: Can integrate with login endpoint now

File: frontend/src/routes/signup/logic.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+export const signUpSchema = z
+  .object({
+    email: z.string().email(),
+    username: z.string().min(2).max(50),
+    firstName: z.string().min(2).max(50),
```
Comment (TabrizPlv) [inline frontend/src/routes/signup/logic.ts]: image
Can we add custom error messages for invalid inputs for a better user experience. Default error message is **String** must contain....
Users may not know what strings are

**Review by TabrizPlv — CHANGES_REQUESTED** [review]
Logic and design looks great. With the exception of minor UI/UX improvements.

File: frontend/src/routes/login/logic.ts (hunk 1 above)
Comment (SeeuSim) [reply frontend/src/routes/login/logic.ts:14]: E2E to be done once full user service is completed, including signing up endpoints as well as username validation.

**Pull request merged**